bin/propolis-utils/src/bin/cpuid-gen.rs

```diff
@@ -0,0 +1,249 @@
+use std::cmp::{Ord, Ordering};
+use std::collections::BTreeMap;
+
+use bhyve_api::{VmmCtlFd, VmmFd};
+use clap::Parser;
+
+fn create_vm() -> anyhow::Result<VmmFd> {
+    let name = format!("cpuid-gen-{}", std::process::id());
+    let mut req = bhyve_api::vm_create_req::new(&name);
+
+    let ctl = VmmCtlFd::open()?;
+    let _ = unsafe { ctl.ioctl(bhyve_api::VMM_CREATE_VM, &mut req) }?;
+
+    let vm = VmmFd::open(&name).map_err(|e| {
+        // Attempt to manually destroy the VM if we cannot open it
+        let mut destroy = bhyve_api::vm_destroy_req::new(&name);
+        let _ = unsafe { ctl.ioctl(bhyve_api::VMM_DESTROY_VM, &mut destroy) };
+        e
+    })?;
+
+    vm.ioctl_usize(bhyve_api::ioctls::VM_SET_AUTODESTRUCT, 1).map_err(|e| {
+        // Destroy instance (and faile) if auto-destruct cannot be set
```

Comment:
nit: don't need the bit in parens there?

Reply:
Will clean that up